core/io/http_client_tcp.cpp

```diff
@@ -506,13 +505,20 @@ Error HTTPClientTCP::poll() {
 						if (s.length() == 0) {
```

Comment:
Thhinking about it, the two lines above could be avoided too, as responses[i] is cleared at the end, so if we add a to_lower based on Spans, the 2 allocations could be avoided. It could be avoided too if we had a `begins_with` case insensitive version, but I don't think we do. @Ivorforce you know the `string` interface well, do you think we can do option 2, or should I addd option 1 ?

Reply:
I think option 2 would be nice. I would probably add `bool String::equalsn_span(const Span<char32_t> &p_lhs , const Span<char32_t> &p_rhs)`, and then
```c++
static bool begins_withn_span(const Span<char32_t> &p_string, const Span<char32_t> &p_span) {
    return p_string.size() >= p_span.size() && equalsn_span(Span(p_string.ptr(), p_span.size()), p_span);
}
```